### PR TITLE
ci: Make superuser opt-in for tests

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
requires some changes in getsentry before landing

File: src/sentry/testutils/cases.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def make_request(self, user=None, auth=None, method=None):
 
     # TODO(dcramer): we want to make the default behavior be ``superuser=False``
```
Comment (dcramer) [inline src/sentry/testutils/cases.py]: need to remove this comment

**Review by mitsuhiko — APPROVED** [review]
Seems fine but is there a reason you are dropping the `format='json'` thing here?

**Review by mattrobenolt — APPROVED** [review]
sounds great

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
might have broken some tests as i changed a bunch of implicit behavior to prevent future bad tests

**Comment by dcramer** [comment]
@mitsuhiko thats actually what the default is -- same reason i removed a bunch of name=xxx values

lots of things arent explicitly testing for defaults or generated outcomes, so just trying to limit the copy pasta code

**Pull request merged**